### PR TITLE
Feature/update shared workflows

### PR DESCRIPTION
**Pull request by marns93**
* Bumped actions to latest versions
* Suggestions: specify default versions for newly introduced configurable language versions (https://github.com/pulumiverse/infra/issues/47)

### CODE REVIEW

File: .github/workflows/provider-build-sdk.yaml [hunk 1]
```diff
@@ -7,17 +7,21 @@ on:
         required: true
         type: string
       dotnetversion:
-        required: true
+        required: false
```
Comment (ringods) [inline .github/workflows/provider-build-sdk.yaml:10]: @marns93 the intent is that consumers of these reusable workflows only use tagged versions, not `latest on main`. Bumping from the first tagged version to the second should be done manually, including all the needed changes to call the new version with the language versions you want in your provider IMO.

Comment (marns93) [reply .github/workflows/provider-build-sdk.yaml:10]: Okay, thanks for the explanation. @ringods 

I agree that using tags is better than using `main`, but I think specifying default versions for the `tags` is easier to use at the end. Only if some provider needs a custom version, once can update it inside the provider.

But it is fine for me to bump the versions in the future myself.

Should we keep this PR the bump the action versions, or should I decline it?

Comment (ringods) [reply .github/workflows/provider-build-sdk.yaml:10]: @marns93 one thing I might not have mentioned for now is that I would like to have workflow maintenance take care of in a very automated way via central pulumiverse workflows. Switching from the version without the language versions, to a version with these, would become an auto-generated PR with this as change.

From:

```yaml
jobs:
  prerequisites:
    uses: pulumiverse/infra/.github/workflows/provider-prerequisites.yaml@actions-workflows-v0.0.1
    with:
      provider: heroku
  build:
    needs: prerequisites
    uses: pulumiverse/infra/.github/workflows/provider-build-sdk.yaml@actions-workflows-v0.0.1
    with:
      provider: heroku
```

to

```yaml
jobs:
  prerequisites:
    uses: pulumiverse/infra/.github/workflows/provider-prerequisites.yaml@actions-workflows-v0.0.2
    with:
      provider: heroku
      dotnetversion: 3.1.301
      goversion: 1.19.x
      nodejsversion: 1.16
      pythonversion: 3.10
  build:
    needs: prerequisites
    uses: pulumiverse/infra/.github/workflows/provider-build-sdk.yaml@actions-workflows-v0.0.2
    with:
      provider: heroku
      dotnetversion: 3.1.301
      goversion: 1.19.x
      nodejsversion: 1.16
      pythonversion: 3.10
```

There will be substantial testing on our end before we would tag a new workflow release and have PRs like above pushed to your repository.

Comment (marns93) [reply .github/workflows/provider-build-sdk.yaml:10]: Okay I see your point and there is no need to push this PR further. Will close it.

Thanks for the explanation @ringods

**Pull request closed without merging**